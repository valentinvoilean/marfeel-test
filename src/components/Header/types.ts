export interface HeaderState {
  isHeaderVisible: boolean;
  isNavigationVisible: boolean;
}
