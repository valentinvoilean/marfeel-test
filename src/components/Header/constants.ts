export const firstWaypoint = 400;
export const secondWaypont = 800;
export const throttleTimeout = 250;
