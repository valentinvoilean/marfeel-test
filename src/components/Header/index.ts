export { HeaderContainer as Header } from './container';
