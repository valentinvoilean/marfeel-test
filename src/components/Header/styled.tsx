import styled from 'styled-components';

export const StyledHeader = styled.header`
  display: flex;
  flex-direction: column;
  background: ${props => props.theme.header.background};
  color: ${props => props.theme.colors.secondaryColor};
  padding: ${props => `${props.theme.sizes.size1} ${props.theme.sizes.size2}`} 0;
`;

export const StyledTopSection = styled.section`
  display: flex;
  flex-wrap: no-wrap;
  align-items: center;
  margin-bottom: ${props => props.theme.sizes.size1};
`;

export const StyledRightSideSection = styled.div`
  display: flex;
  flex-grow: 1;
  justify-content: center;
`;
