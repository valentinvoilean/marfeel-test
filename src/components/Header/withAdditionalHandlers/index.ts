export * from './withAdditionalHandlers';
export * from './types';
