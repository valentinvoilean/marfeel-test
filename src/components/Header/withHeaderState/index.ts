export * from './types';
export * from './withHeaderState';
