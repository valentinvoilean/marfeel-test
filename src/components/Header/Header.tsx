import throttle from 'lodash.throttle';
import React from 'react';
import LoadingBar from 'react-redux-loading-bar';

import { StyledHeader, StyledRightSideSection, StyledTopSection } from './styled';

import { Logo } from '../shared/Logo';
import { BurgerMenu } from './components/BurgerMenu';
import { Navigation } from './components/Navigation';

import { HeaderState } from './types';

import { firstWaypoint, secondWaypont, throttleTimeout } from './constants';

export class Header extends React.PureComponent<{}, HeaderState> {
  public state = {
    isNavigationVisible: window.scrollY <= firstWaypoint,
    isHeaderVisible: window.scrollY <= secondWaypont
  };

  private checkTopOffset = throttle(() => {
    const { isNavigationVisible, isHeaderVisible } = this.state;
    const newNavigationState = window.scrollY <= firstWaypoint;
    const newHeaderState = window.scrollY <= secondWaypont;

    if (isHeaderVisible !== newHeaderState || isNavigationVisible !== newNavigationState) {
      this.setState({
        isNavigationVisible: newNavigationState,
        isHeaderVisible: newHeaderState
      });
    }
  }, throttleTimeout);

  public componentDidMount(): void {
    window.addEventListener('scroll', this.checkTopOffset);
  }

  public componentWillUnmount(): void {
    window.removeEventListener('scroll', this.checkTopOffset);
  }

  public render(): JSX.Element {
    const { isHeaderVisible, isNavigationVisible } = this.state;

    if (!isHeaderVisible) {
      return <BurgerMenu showWrapper={true} />;
    }

    return (
      <StyledHeader>
        <LoadingBar />
        <StyledTopSection>
          <BurgerMenu />
          <StyledRightSideSection>
            <Logo />
          </StyledRightSideSection>
        </StyledTopSection>
        {isNavigationVisible && <Navigation />}
      </StyledHeader>
    );
  }
}
