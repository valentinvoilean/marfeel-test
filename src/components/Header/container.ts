import { compose, pure } from 'recompose';

import { withAdditionalHandlers } from './withAdditionalHandlers';
import { withHeaderStateHandlers } from './withHeaderState';
import { withLifecycleMethods } from './withLifecycleMethods';

import { Header } from './Header';

import { ComposedProps } from './types';

export const HeaderContainer = compose<ComposedProps, {}>(
  withHeaderStateHandlers,
  withAdditionalHandlers,
  withLifecycleMethods,
  pure
)(Header);
