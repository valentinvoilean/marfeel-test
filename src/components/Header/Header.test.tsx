import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-styled-components';

import { Header } from './Header';

import { StoreStyledProvider } from '../shared/StoreStyledProvider';

describe('Header', () => {
  it('renders without crashing', () => {
    const header = renderer
      .create(
        <StoreStyledProvider>
          <Header />
        </StoreStyledProvider>
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
