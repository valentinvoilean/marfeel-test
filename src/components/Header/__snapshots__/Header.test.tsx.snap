// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  top: 0;
  width: 100%;
  background: rgb(2,0,36);
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 15px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  width: 100px;
  height: 25px;
}

.c2 {
  border: none;
  padding: 0;
  -webkit-appearance: unset;
  background: none;
  height: 16px;
  width: 20px;
  position: absolute;
  cursor: pointer;
}

.c3 {
  position: absolute;
  width: 20px;
  height: 2px;
  border-radius: 2px;
  background-color: white;
  top: 7px;
}

.c3:before,
.c3:after {
  display: block;
  content: '';
  position: absolute;
  width: 20px;
  height: 2px;
  border-radius: 2px;
  background-color: white;
}

.c3:before {
  top: -6px;
}

.c3:after {
  bottom: -6px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  height: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  font-weight: 400;
}

.c7 {
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 0 15px;
  margin: 0;
  overflow: auto;
  height: 50px;
}

.c8 {
  white-space: nowrap;
  padding-right: 15px;
  color: white;
}

.c9 {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding-bottom: 2px;
  border-bottom: 2px solid white;
}

.c10 {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding-bottom: 2px;
}

<header
  className="c0"
>
  <div />
  <section
    className="c1"
  >
    <button
      className="c2"
      onClick={[Function]}
    >
      <div
        className="c3"
      />
    </button>
    <div
      className="c4"
    >
      <img
        alt="Logo"
        className="c5"
        src="marfeel_logo.svg"
      />
    </div>
  </section>
  <nav
    className="c6"
  >
    <ul
      className="c7"
    >
      <li
        className="c8"
      >
        <a
          className="c9"
          href="#1"
        >
          Section 1
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#2"
        >
          Section 2
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#3"
        >
          Section 3
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#4"
        >
          Section 4
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#5"
        >
          Section 5
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#6"
        >
          Section 6
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#7"
        >
          Section 7
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#8"
        >
          Section 8
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#9"
        >
          Section 9
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#10"
        >
          Section 10
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
